containers/unit-tests/semaphore-cache

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+set -eux
+
+item="$2"
+item_esc="${item//\//_}"
+item_esc="${item_esc//:/_}"
+
+case "${1:-}" in
+    docker-load)
+        for f in $SEMAPHORE_CACHE_DIR/docker/$item_esc-*; do
+            [ -e "$f" ] || continue
+            echo "Restoring cached Docker image $f..."
+            docker load < "$f"
+        done
+
+        if [ -d "$SEMAPHORE_CACHE_DIR/docker" ]; then
+            echo "Cleaning old cached docker images..."
+            find "$SEMAPHORE_CACHE_DIR/docker" -type f -mtime +7 -print -delete
```

Comment:
Wait, this is not sufficient now. This will only cover the "temporary tag now unused" case, but with the standard "update a tag" workflow this would unpack a recent and any older image for a week, the latter in vain. We *do* need to remove all other images immediately after saving, like in my original version. But we can keep that in addition for the former case.

Reply:
I think I was a bit unclear on my proposal for cache invalidation.  My idea was to do the following two independent things, *always*:

 1) delete all hashes of our image name, except for our own hash
 2) delete absolutely everything that wasn't touched in over a week